src/main/java/domain/Names.java

```diff
@@ -1,29 +1,35 @@
 package domain;
 
-import util.ExceptionMessages;
-
-import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
-import java.util.stream.Stream;
+import java.util.stream.Collectors;
 
 public class Names {
 
+    public static final String INVALID_NAMES_COUNT = "이름은 2개 이상 10개 이하 이어야 합니다.";
+    public static final String DUPLICATED_NAMES = "참가자의 이름은 중복될 수 없습니다.";
     private static final int MINIMUM_PLAYERS_COUNT = 2;
     private static final int MAXIMUM_PLAYERS_COUNT = 10;
 
-    private final List<Name> values = new ArrayList<>();
+    private final List<Name> values;
 
     public Names(final List<String> values) {
         validateNameCount(values.size());
-        for (final String name : values) {
-            this.values.add(new Name(name));
-        }
+        validateDuplicatedNames(values);
+        this.values = values.stream()
+                .map(Name::new)
+                .collect(Collectors.toList());
```

Comment:
```suggestion
        this.values = values.stream()
                .map(Name::new)
                .toList();
```
Stream을 활용하면 unmodifiableList를 반환받아볼 수 있습니다~! 한번 들어가서 확인해보시죠

Reply:
toList()가 unmodifiableList로 반환한다는 걸 알고있었는데, 자동 완성에 너무 의존했던 것 같아요!
항상 사용 가능한 메서드가 무엇이 있고, 어떤 것이 적합할지 생각하고 사용하도록 할게요.👍